src/wp-includes/media.php

```diff
@@ -1958,6 +1981,97 @@ function wp_img_tag_add_loading_attr( $image, $context ) {
 	return $image;
 }
 
+/**
+ * Adds optimization attributes to an `img` HTML tag.
+ *
+ * @since 6.3.0
+ *
+ * @param string $image   The HTML `img` tag where the attribute should be added.
+ * @param string $context Additional context to pass to the filters.
+ *
+ * @return string Converted `img` tag with optimization attributes added.
+ */
+function wp_img_tag_add_loading_optimization_attrs( $image, $context ) {
+	$image_elm = new WP_HTML_Tag_Processor( $image );
+	$image_elm->next_tag();
```

Comment:
Just to be safe, we should probably ensure this method provides the expected result and bail otherwise.
```suggestion
	if ( ! $image_elm->next_tag() ) {
		return $image;
	}

```